### PR TITLE
Fix warnings and errors on Windows/MSVC

### PR DESCRIPTION
**Pull request by bobbymcr**
- Provide implementation of strlcat
- Address alloc function warnings by checking for NULL returns
- Fix error C2036 by casting void* to char* before increment

### CODE REVIEW

File: bdd-for-c.h [hunk 1]
```diff
@@ -41,6 +41,23 @@ SOFTWARE.
 #include <Windows.h>
 #include <io.h>
 #define __BDD_IS_ATTY__() _isatty(_fileno(stdout))
+size_t strlcat(char* dst, const char* src, size_t size)
```
Comment (grassator) [inline bdd-for-c.h]: It looks like this is the only dependency we drag from BSD headers on *nix systems so probably worth it to rename this to `__bdd_strlcat__`, compile unconditionally and use it everywhere. Could you please make this change?

File: array.h [hunk 2]
```diff
@@ -9,6 +9,9 @@ typedef struct _bdd_array {
 
 _bdd_array* _bdd_array_create() {
     _bdd_array *arr = malloc(sizeof(_bdd_array));
+    if (!arr) {
+        return NULL;
```
Comment (grassator) [inline array.h]: I think if we can not `malloc` we should just exit with a non-zero status code. Maybe also with a message, although I'm not sure you can output to console it malloc is failing...

File: bdd-for-c.h [hunk 3]
```diff
@@ -496,7 +536,7 @@ for(\
             )\
         )\
     );\
-    ++__bdd_index__\
+    __bdd_index__ = (char*)__bdd_index__ + 1\
```
Comment (grassator) [inline bdd-for-c.h:552]: Why a cast is necessary here?

**Review by grassator — COMMENTED** [review]
Thank you for the PR please take a look at the comments and see if they make sense

File: array.h (hunk 2 above)
Comment (bobbymcr) [reply array.h]: I think perror + abort would be okay here -- worst case, you don't see a message and the process still aborts.

File: bdd-for-c.h (hunk 3 above)
Comment (bobbymcr) [reply bdd-for-c.h:552]: Arithmetic on void* is a [nonstandard GCC extension][1] -- fails with error [C2036][2] in VC.

[1]: https://stackoverflow.com/questions/37460579/error-c2036-void-unknown-size
[2]: https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2036?view=vs-2019

**Comment by grassator** [comment]
Thanks a lot for the contribution!

**Pull request merged**